src/ripple/shamap/impl/SHAMapInnerNode.cpp

```diff
@@ -326,7 +327,8 @@ SHAMapInnerNode::setChild(int m, intr_ptr::SharedPtr<SHAMapTreeNode> child)
         auto const childIndex = *getChildIndex(m);
         auto [_, hashes, children] = hashesAndChildren_.getHashesAndChildren();
         hashes[childIndex].zero();
-        children[childIndex] = std::move(child);
+        children[childIndex].assign(
```

Comment:
I'm still banging my head against the rippled SHAMap, as I have little experience with their "top-down" use for syncing - another world of headache.

If you only enter from the top, seems you're gold with CoW, but it seems like rippled needs to modify a tree while it's in use for syncing purposes? So you need kind of write-only nodes for a time? In rippled are the syncing threads partitioned to certain keyspaces?